src/modules/rppt_tensor_geometric_augmentations.cpp

```diff
@@ -721,6 +721,182 @@ RppStatus rppt_resize_crop_mirror_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** pixelate ********************/
+
+RppStatus rppt_pixelate_host(RppPtr_t srcPtr,
+                             RpptDescPtr srcDescPtr,
+                             RppPtr_t dstPtr,
+                             RpptDescPtr dstDescPtr,
+                             RpptROIPtr roiTensorPtrSrc,
+                             RpptRoiType roiType,
+                             rppHandle_t rppHandle)
+{
+    RppLayoutParams srcLayoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::U8))
+    {
+        RpptImagePatchPtr internalDstImgSizes = (RpptImagePatch *) calloc(dstDescPtr->n, sizeof(RpptImagePatch));
+        RpptROI *internalRoiTensorPtrSrc = (RpptROI *) calloc(dstDescPtr->n, sizeof(RpptROI));
+        for(int i=0;i<dstDescPtr->n;i++)
+        {
+            internalDstImgSizes[i].width = internalRoiTensorPtrSrc[i].xywhROI.roiWidth = roiTensorPtrSrc[i].xywhROI.roiWidth / 8;
+            internalDstImgSizes[i].height = internalRoiTensorPtrSrc[i].xywhROI.roiHeight = roiTensorPtrSrc[i].xywhROI.roiHeight / 8;
+        }
+        RpptDescPtr interDescPtr = dstDescPtr;
+        unsigned long long interBufferSize = (unsigned long long)interDescPtr->h * (unsigned long long)interDescPtr->w * (unsigned long long)dstDescPtr->c * (unsigned long long)dstDescPtr->n;
+        Rpp8u *interDstPtr = (Rpp8u *)calloc(interBufferSize, sizeof(Rpp8u));
+        resize_bilinear_u8_u8_host_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                                          srcDescPtr,
+                                          static_cast<Rpp8u*>(interDstPtr),
+                                          interDescPtr,
+                                          internalDstImgSizes,
+                                          roiTensorPtrSrc,
+                                          roiType,
+                                          srcLayoutParams);
+
+        for(int i=0;i<dstDescPtr->n;i++)
+        {
+            internalRoiTensorPtrSrc[i].xywhROI.xy.x = roiTensorPtrSrc[i].xywhROI.xy.x / 8;
```

Comment:
Is width / 8 and height / 8 fixed?
Can it change to width / n or height / n (n = any number)?

Reply:
Yeah, we can change n based on amount of pixelation required.
In BatchPD they had used kernelsize of 7 to compute average,  however using 8 as resize ratio lead to outputs closer to BatchPD.

I will add a variable named kernelsize and set it to 8 by default.